Tests/SwaggerParserTests/SwaggerParserTests.swift

```diff
@@ -1,20 +1,159 @@
-
 class SwaggerParserTests: XCTestCase {
-    func testUber() {
-        do {
-            let swagger = try Swagger(JSONString: uberJSONString)
-            print(swagger)
-        } catch {
-            print(error)
+    var testFixtureFolder: URL!
+    
+    override func setUp() {
+        testFixtureFolder = URL(fileURLWithPath: #file).deletingLastPathComponent().appendingPathComponent("Fixtures")
+    }
+    
+    func testInitialization() throws {
+        let jsonString = try self.fixture(named: "uber.json")
+        let swagger = try Swagger(JSONString: jsonString)
+        
+        XCTAssertEqual(swagger.host?.absoluteString, "api.uber.com")
+    }
+    
+    func testAllOfSupport() throws {
+        let jsonString = try fixture(named: "test_all_of.json")
+        let swagger = try Swagger(JSONString: jsonString)
+        
+        guard
+            let baseDefinition = swagger.definitions.first(where: {$0.name == "TestAllOfBase"}),
+            case .object(let baseSchema) = baseDefinition.structure else
+        {
+            XCTFail("TestAllOfBase is not an object schema."); return
+        }
+        
+        validate(testAllOfBaseSchema: baseSchema)
+        try validate(thatDefinitions: swagger.definitions, containsTestAllOfChild: "TestAllOfFoo", withPropertyNames: ["foo"])
+        try validate(thatDefinitions: swagger.definitions, containsTestAllOfChild: "TestAllOfBar", withPropertyNames: ["bar"])
+    }
+}
+
+/// MARK: Helper Functions
+
+fileprivate extension SwaggerParserTests {
+    func fixture(named fileName: String) throws -> String {
+        let url = testFixtureFolder.appendingPathComponent(fileName)
+        return try String.init(contentsOf: url, encoding: .utf8)
+    }
+}
+    
+fileprivate enum GetBaseAndChildSchemasError: Error {
+    case missingBase
+    case missingChild
+    case badSubschemaType(Schema)
+}
+
+/// Gets the base schema and child schema from a definition that defines an
+/// `allOf` with one $ref (the base class) and one object schema.
+fileprivate func getBaseAndChildSchemas(withDefinition definition: Structure<Schema>) throws -> (base: ObjectSchema, child: ObjectSchema) {
+    var base: ObjectSchema!
+    var child: ObjectSchema!
+    
+    if case .allOf(let allOfSchema) = definition.structure {
```

Comment:
How about turning this into a guard statement that throws an error. Then you don't have to indent the rest of this if statement. I.e.:

```
guard case .allOf(let allOfSchema) = definition.structure else {
    throw GetBaseAndChildSchemasError.notAllOf
}

XCTAssertEqual(allOfSchema.subschemas.count, 2)

var base: ObjectSchema?
var child: ObjectSchema?

try allOfSchema.subschemas.forEach { subschema in
    switch subschema {
    case .object(let childSchema):
        child = childSchema
    case .structure(let structure):
        if case .object(let baseSchema) = structure.structure {
            base = baseSchema
        } else {
            throw GetBaseAndChildSchemasError.badSubschemaType(subschema)
        }
    default:
        throw GetBaseAndChildSchemasError.badSubschemaType(subschema)
    }
}

if let base = base, let child = child {
    return (base: base, child: child)
}

if base == nil {
    throw GetBaseAndChildSchemasError.missingBase
} else {
    throw GetBaseAndChildSchemasError.missingChild
}